doc/reference/networking/openthread_integration.rst

```diff
@@ -0,0 +1,172 @@
+Each of these flows is described below in an appropriate section.
+
+RX path
+*******
+An application typically consists of one or more :ref:`threads <threads_v2>`
+that execute the application logic. When using the
+:ref:`BSD socket API <bsd_sockets_interface>`, the following things will
+happen.
+
+.. figure:: zephyr_netstack_openthread-rx_sequence.svg
+    :alt: OpenThread Application RX data flow
+    :figclass: align-center
+
+    OpenThread Application RX data flow
+
+Data receiving (RX) - TODO: expand, remove/rephrase copy/paste from the zephyrs RX page.
+-------------------
+
+1. A network data packet is received by a nRF5 IEEE 802.15.4 Radio Driver.
+
+2. The device driver allocates enough network buffers to store the received
+   data. The network packet is placed in the proper RX queue (implemented by
+   :ref:`k_fifo <fifos_v2>`). By default there is only one receive queue in
+   the system, but it is possible to have up to 8 receive queues.
+   These queues will process incoming packets with different priority.
+   See :ref:`traffic-class-support` for more details. The receive queues also
+   act as a way to separate the data processing pipeline (bottom-half) as
+   the device driver is running in an interrupt context and it must do its
+   processing as fast as possible.
+
+3. The 802154 RX Radio driver thread does the upper-half processing of the
+   received IEEE 802.15.4 radio frame. As a result it puts a work item
+   with the net_recv_data() to have the frame processed.
+
+4. The work queue thread rx_workq calls the registered handler for every queued frame.
+   In this case the registered handler openthread_recv() checks if the frame is of the 
+   IEEE 802.15.4 type and if it is the case it insertd the frame in the rx_pkt_fifo and returns the NET_OK.
+
+5. The OpenThread thread gets a frame from the FIFO and processes it.
+   It also handles IP header compression, reassembly of fragmented traffic.
+
+6. As soon as it detects a valid IPv6 packet that needs to be handled by the 
+   higher layer it calls the registered callback ot_receive_handler()
+   which creates a buffer for a net_pkt that is going to be passed to the Zephyr's IP stack
+   and calls the net_recv_data() to have it processed.
+
+7. This time the openthread_recv() called by the work queue returns NET_CONTINUE
+   indicating that the valid IPv6 packet is present and needs to be processed by
+   the Zephyr's higher layer.
+
+8. The net_ipv6_input() passes the packet the next higher layer.
+
+9. The packet is then passed to L3 processing. If the packet is IP based,
+   then the L3 layer processes the IPv6.
+
+10. A socket handler then finds an active socket to which the network packet
+   belongs and puts it in a queue for that socket, in order to separate the
+   networking code from the application. Typically the application is run in
+   userspace context and the network stack is run in kernel context.
```

Comment:
I'm not sure anymore we should mention userspace here as we don't have thread app that separates userspace from kernelspace.

Reply:
Removed both from images as well.